packages/api-server/api_server/repositories/alerts.py

```diff
@@ -1,94 +1,234 @@
 import logging
+from collections import deque
 from datetime import datetime
 from typing import List, Optional
 
-from fastapi import Depends
-
-from api_server.authenticator import user_dep
-from api_server.models import User
+from api_server.models import AlertRequest, AlertResponse
 from api_server.models import tortoise_models as ttm
-from api_server.repositories.tasks import TaskRepository
 
+# TODO: not hardcode all these expected values
+LocationAlertSuccessResponse = "success"
+LocationAlertFailResponse = "fail"
+LocationAlertTypeParameterName = "type"
+LocationAlertTypeParameterValue = "location_result"
+LocationAlertLocationParameterName = "location_name"
+LocationAlertFinalCheckTypeParameterValue = "check_all_task_location_alerts"
 
-class AlertRepository:
-    def __init__(
-        self,
-        user: User = Depends(user_dep),
-        task_repo: TaskRepository = Depends(TaskRepository),
+
+def get_location_from_location_alert(alert: AlertRequest) -> Optional[str]:
+    """
+    Returns the location name from a location alert when possible, otherwise
+    returns None.
+    Note: This is an experimental feature and may be subjected to
+    modifications often.
+    """
+    if (
+        len(alert.alert_parameters) < 2
+        or LocationAlertSuccessResponse not in alert.responses_available
+        or LocationAlertFailResponse not in alert.responses_available
     ):
-        self.user = user
-        self.task_repo = task_repo
+        return None
 
-    async def get_all_alerts(self) -> List[ttm.AlertPydantic]:
-        alerts = await ttm.Alert.all()
-        return [await ttm.AlertPydantic.from_tortoise_orm(a) for a in alerts]
+    # Check type
+    alert_type = None
+    for param in alert.alert_parameters:
+        if param.name == LocationAlertTypeParameterName:
```

Comment:
Generally better to not mix the type identifier with the rest of the parameters.